services/pipenv-helpers.js

```diff
@@ -0,0 +1,49 @@
+'use strict'
+
+const Joi = require('@hapi/joi')
+const { InvalidParameter } = require('.')
+
+const isDependency = Joi.object({
+  version: Joi.string().required(),
+}).required()
+
+const isLockfile = Joi.object({
+  _meta: Joi.object({
+    requires: Joi.object({
+      python_version: Joi.string().required(),
+    }).required(),
+  }).required(),
+  default: Joi.object().pattern(Joi.string().required(), isDependency),
+  develop: Joi.object().pattern(Joi.string().required(), isDependency),
+}).required()
+
+function getDependencyVersion({
+  kind = 'prod',
+  wantedDependency,
+  lockfileData,
+}) {
+  let dependenciesOfKind
+  if (kind === 'dev') {
+    dependenciesOfKind = lockfileData.develop
+  } else if (kind === 'prod') {
+    dependenciesOfKind = lockfileData.default
+  } else {
+    throw Error(`Not very kind: ${kind}`)
```

Comment:
This seems like a bit of an odd error. I'd be confused by it if it popped up in Sentry :thinking:

Reply:
Heh. Copy-pasted from package-json-helpers. There was some chat about it then; basically it's an assertion failure which is unlikely ever to happen. Can change it to something more specific though.

Reply:
This would only happen in the event of a programming error, correct?